src/macaron/build_spec_generator/cli_command_parser/maven_cli_parser.py

```diff
@@ -0,0 +1,594 @@
+            The MavenCLICommand instance.
+
+        Raises
+        ------
+        MavenCLICommandParseError
+            If an error happens when parsing the Maven CLI Command.
+        """
+        if not cmd_list:
+            raise CommandLineParseError("The provided cmd list is empty.")
+
+        exe_path = cmd_list[0]
+        options = cmd_list[1:]
+
+        if os.path.basename(exe_path) not in MavenCLICommandParser.ACCEPTABLE_EXECUTABLE:
+            raise CommandLineParseError(f"{exe_path} is not an acceptable mvn executable path.")
+
+        # TODO: because our parser is not completed for all cases, should we be more relaxed and use
+        # parse_unknown_options?
+        try:
+            parsed_opts = self.arg_parser.parse_args(options)
+        except argparse.ArgumentError as error:
+            raise CommandLineParseError(f"Failed to parse command {' '.join(options)}.") from error
+        # Even though we have set `exit_on_error`, argparse still exists unexpectedly in some
+        # cases. This has been confirmed to be a bug in the argparse library implementation.
+        # https://github.com/python/cpython/issues/121018.
+        # This is fixed in Python3.12, but not Python3.11
+        except SystemExit as sys_exit_err:
+            raise CommandLineParseError(f"Failed to parse the Maven CLI Options {' '.join(options)}.") from sys_exit_err
+
+        # Handle cases where goal or plugin phase is not provided.
+        if not parsed_opts.goals:
+            # Allow cases such as:
+            #   mvn --help
+            #   mvn --version
+            # Note that we don't allow mvn -V or mvn --show-version as this command will
+            #   failed for mvn
+            if not parsed_opts.help_ and not parsed_opts.version:
+                raise CommandLineParseError(f"No goal detected for {' '.join(options)}.")
+
+        maven_cli_options = MavenCLIOptions.from_parsed_arg(parsed_opts)
+
+        return MavenCLICommand(
+            executable=exe_path,
+            options=maven_cli_options,
+        )
+
+    def _patch_properties_mapping(
+        self,
+        original_props: dict[str, str],
+        option_long_name: str,
+        patch_value: MavenOptionPatchValueType,
+    ) -> dict[str, str]:
+        define_opt_def = self.option_defs.get(option_long_name)
+        if not define_opt_def or not isinstance(define_opt_def, MavenSystemPropeties):
+            raise PatchBuildCommandError(f"{option_long_name} from the patch is not a --define option.")
+
+        if not define_opt_def.is_valid_patch_option(patch_value):
+            raise PatchBuildCommandError(f"Critical, incorrect runtime type for patch --define, value: {patch_value}.")
+
+        return patch_mapping(
+            original=original_props,
+            patch=patch_value,
+        )
+
+    def apply_patch(
+        self,
+        cli_command: MavenCLICommand,
+        options_patch: Mapping[str, MavenOptionPatchValueType | None],
```

Comment:
```suggestion
        patch_options: Mapping[str, MavenOptionPatchValueType | None],
```
I think this is actually more appropriate than `options_patch`. Also applies to the Gradle counterpart.